src/formatDate.js

```diff
@@ -8,7 +8,85 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  // find out which separator
+  // to split the string
+  // to convert to an array of numbers
+  const pastSeparator = fromFormat[3];
+  const oldFormat = date.split(pastSeparator);
+
+  let year = '';
+  let month = '';
+  let day = '';
+
+  // find out where the year, month and day
+  // are in the old date format
+  // assign appropriate values to the variables
+
+  // don't take the separator (the last element of the fromFormat array)
+  // for loop
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    switch (fromFormat[i]) {
+      case 'YYYY':
+      case 'YY':
+        year += oldFormat[i];
+        break;
+
+      case 'DD':
+        day += oldFormat[i];
+        break;
+
+      case 'MM':
+        month += oldFormat[i];
```

Comment:
Using '+=' is unnecessary here because you're not appending to the 'month' but rather assigning a new value. Use '=' instead.